scopesim/effects/data_container.py

```diff
@@ -56,16 +56,24 @@ class DataContainer:
         to the data open FITS file.
 
     """
-
     def __init__(self, filename=None, table=None, array_dict=None, **kwargs):
 
         if filename is None and "file_name" in kwargs:
             filename = kwargs["file_name"]
 
         filename = utils.find_file(filename)
         self.meta = {"filename": filename,
```

Comment:
Pensive: Perhaps you could make `self.meta` a dataclass with defaults? Then you don't have to do `if '...' in self.meta` etc.